### PR TITLE
#2566 Improve DynaFlow SVC

### PR DESCRIPTION
**Pull request by cossartque**
**Checklist before requesting a review**

*use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes*

- [ ] unit tests and non regression tests were added (new model, new feature and bug fix)
- [ ] main documentation was updated (update of input/output file, 3rd party, model, repository organization, solver)
- [ ] example documentations were updated (new example in examples folder)
- [ ] if this PR modifies the parameters or inputs/outputs of a model/solver: the corresponding xsl was added in util/xsl and platform DB were updated
- [ ] if this PR modifies a dictionary: the corresponding french dictionary was updated
- [ ] if this commit targets a release branch: the corresponding milestone was added in the ticket and in this PR

### CODE REVIEW

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/SecondaryVoltageControl/Simplified/SecondaryVoltageControl_INIT.mo [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite of simulation tools for power systems.
+*/
+
+model SecondaryVoltageControl_INIT "Initialisation model for simplified secondary voltage control"
+  import Dynawo.Electrical.SystemBase;
+  import Dynawo.Types;
+
+  parameter Integer NbMaxGen = 50 "Maximum number of generators that should participate in the considered secondary voltage control";
+  parameter Boolean[NbMaxGen] Participate0 = fill(false, NbMaxGen) "If true, the generator participates in the secondary voltage control. If false, the generator does not participate in the secondary voltage control (for each generator connected to the SVC).";
```
Comment (ErwanGuichard49) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/SecondaryVoltageControl/Simplified/SecondaryVoltageControl_INIT.mo:20]: You can remove the "If false ..." part from the comment : it is simply the opposite of the "If true ..." part.

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/SecondaryVoltageControl/Simplified/SecondaryVoltageControl_INIT.mo [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  parameter Types.ReactivePower[NbMaxGen] Qr = zeros(NbMaxGen) "Participation factor of the generators to the secondary voltage control in Mvar (for each generator connected to the SVC)";
+
+  Types.PerUnit Level0 "Initial level demand";
+  Integer[NbMaxGen] Participate0Int "If 1, the generator participates in the secondary voltage control. If 0, the generator does not participate in the secondary voltage control (for each generator connected to the SVC).";
+
+equation
+  for i in 1:NbMaxGen loop
+    Participate0Int[i] = if Participate0[i] then 1 else 0;
+  end for;
+  Level0 = sum(-Q0Pu .* SystemBase.SnRef .* Participate0Int) / sum(Qr .* Participate0Int);
```
Comment (ErwanGuichard49) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/SecondaryVoltageControl/Simplified/SecondaryVoltageControl_INIT.mo:31]: I would write :

Level0 = -SystemBase.SnRef * sum(-Q0Pu .* Participate0Int) / sum(Qr .* Participate0Int);

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/SecondaryVoltageControl/Simplified/SecondaryVoltageControl.mo [hunk 3]
```diff
@@ -17,35 +17,35 @@ model SecondaryVoltageControl "Model for simplified secondary voltage control"
   import Dynawo.Types;
 
   //Regulation parameters
-  parameter Types.PerUnit Ck "Sensitivity of voltage to level variations in pu (base UNom)";
+  parameter Types.Time Alpha "PI integral gain";
```
Comment (ErwanGuichard49) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/SecondaryVoltageControl/Simplified/SecondaryVoltageControl.mo:20]: If this is a time constant, shouldn't you choose a name beginning by t ?

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/SecondaryVoltageControl/Simplified/SecondaryVoltageControl.mo [hunk 4]
```diff
@@ -17,35 +17,35 @@ model SecondaryVoltageControl "Model for simplified secondary voltage control"
...
+  parameter Types.Time Beta "PI proportional gain";
```
Comment (ErwanGuichard49) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/SecondaryVoltageControl/Simplified/SecondaryVoltageControl.mo:21]: Same remark

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/SecondaryVoltageControl/Simplified/SecondaryVoltageControl.mo (hunk 3 above)
Comment (cossartque) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/SecondaryVoltageControl/Simplified/SecondaryVoltageControl.mo:20]: It is not a time constant in fact, I change the type.

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/SecondaryVoltageControl/Simplified/SecondaryVoltageControl_INIT.mo (hunk 1 above)
Comment (cossartque) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/SecondaryVoltageControl/Simplified/SecondaryVoltageControl_INIT.mo:20]: done

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/SecondaryVoltageControl/Simplified/SecondaryVoltageControl_INIT.mo (hunk 2 above)
Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/SecondaryVoltageControl/Simplified/SecondaryVoltageControl_INIT.mo:31]: I understand.

**Review by ErwanGuichard49 — APPROVED** [review]

**Comment by cossartque** [comment]
@rosiereflo fyi, you can see in the .par in the example the parameters that we now need for the SVC. We have generic parameters (Alpha, Beta, DerLevelMaxPu) and init values (Up0Pu, UpRef0Pu (they are both equal to the u of the pilot point in the iidm)). We also have parameters and init values for each generator of the SVC zone (P0Pu, Q0Pu, U0Pu, Participate0, SNom, Qr (given by DynaWaltz DB), XTfoPu if there is a Tfo). We have a max size of 50 but that doesn't mean we need to give parameters for the 50 generators as we have default values. If a generator is in the zone in the DB but not present in the situation, giving its parameters or not won't change anything (SNom, XTfoPu, Qr) as Participate0 is by default false.

**Comment by cossartque** [comment]
@marcochiaramello this branch is now ready for review and merge, thanks

**Pull request merged**